solidity/test/TestBeaconBackportRewards.js

```diff
@@ -0,0 +1,191 @@
+        expect(await rewards.lastInterval()).to.eq.BN(1);
+    })
+
+    it("should exclude group 1", async () => {
+        assert.isTrue(await rewards.isExcluded(1), "group 1 not excluded")
+    })
+
+    it("should recognize groups 0~3", async () => {
+        let recognized0 = await rewards.recognizedByFactory(0);
+        let recognized1 = await rewards.recognizedByFactory(1);
+        let recognized2 = await rewards.recognizedByFactory(2);
+        let recognized3 = await rewards.recognizedByFactory(3);
+        let recognized4 = await rewards.recognizedByFactory(4);
+
+        assert.isTrue(recognized0, "group 0 not recognized")
+        assert.isTrue(recognized1, "group 1 not recognized")
+        assert.isTrue(recognized2, "group 2 not recognized")
+        assert.isTrue(recognized3, "group 3 not recognized")
+        assert.isFalse(recognized4, "group 4 falsely recognized")
+    })
+
+    it("should have groups 0, 2 and 3 eligible", async () => {
+        let eligible0 = await rewards.eligibleForReward(0);
+        let eligible1 = await rewards.eligibleForReward(1);
+        let eligible2 = await rewards.eligibleForReward(2);
+        let eligible3 = await rewards.eligibleForReward(3);
+
+        assert.isTrue(eligible0, "group 0 ineligible")
+        assert.isFalse(eligible1, "group 1 eligible")
+        assert.isTrue(eligible2, "group 2 ineligible")
+        assert.isTrue(eligible3, "group 3 ineligible")
+    })
+
+    it("should recognize group 1 as terminated", async () => {
+        let terminated0 = await rewards.eligibleButTerminatedWithUint(0);
+        let terminated1 = await rewards.eligibleButTerminatedWithUint(1);
+        let terminated2 = await rewards.eligibleButTerminatedWithUint(2);
+        let terminated3 = await rewards.eligibleButTerminatedWithUint(3);
+
+        assert.isFalse(terminated0, "group 0 falsely terminated")
+        assert.isTrue(terminated1, "group 1 not terminated")
+        assert.isFalse(terminated2, "group 2 falsely terminated")
+        assert.isFalse(terminated3, "group 3 falsely terminated")
+    })
+
+    it("should register 3 groups in the first interval", async () => {
+        let count = await rewards.findEndpoint(await rewards.endOf(0))
+        expect(count).to.eq.BN(3)
+    })
+
+    it("should register 1 group in the second interval", async () => {
+        let count = await rewards.findEndpoint(await rewards.endOf(1))
+        expect(count).to.eq.BN(4)
+    })
+
+    it("should receive rewards for groups 0, 2 and 3", async () => {
+        await rewards.receiveReward(0);
+
+        let balance1 = await token.balanceOf(beneficiary1);
+        expect(balance1).to.eq.BN(500)
+
+        await rewards.receiveReward(2);
+        let balance2 = await token.balanceOf(beneficiary2);
+        expect(balance2).to.eq.BN(2000)
+
+        await rewards.receiveReward(3);
+        let balance1new = await token.balanceOf(beneficiary1)
+        expect(balance1new).to.eq.BN(2500)
```

Comment:
I'd love to make this test a little more explanatory, it took me a while to figure out where those balances come from. My proposition:
```
        // 4500 allocated to the first interval:
        //   1500 allocated to group 0
        //   1500 not allocated to group 1 because it's terminated
        //   1500 allocated to group 2
        // 4500 allocated to the second interval:
        //   4500 allocated to group 3

        // 1500 allocated to group 0
        await rewards.receiveReward(0);
        expect(await token.balanceOf(beneficiary1)).to.eq.BN(500)
        expect(await token.balanceOf(beneficiary2)).to.eq.BN(1000)

        // 1500 allocated to group 2
        await rewards.receiveReward(2);
        expect(await token.balanceOf(beneficiary1)).to.eq.BN(1000) // 500+500
        expect(await token.balanceOf(beneficiary2)).to.eq.BN(2000) // 1000+1000

        // 4500 allocated to group 3
        await rewards.receiveReward(3);
        expect(await token.balanceOf(beneficiary1)).to.eq.BN(2500) // 1000+1500
        expect(await token.balanceOf(beneficiary2)).to.eq.BN(5000) // 2000+3000
```